comments/views.py

```diff
@@ -0,0 +1,69 @@
+from django.db.models import Q
+from django.contrib import messages
+from django.shortcuts import redirect
+from django.views import View
+from django.views.decorators.csrf import csrf_protect
+from django.views.decorators.http import require_POST
+from django.views.generic import FormView, TemplateView
+from django_comments.views.comments import post_comment
+from django_comments_xtd import get_form
+from django_comments_xtd.models import XtdComment
+from django.utils.translation import ugettext as _
+
+from comments.forms import CommentForm, AdminCommentForm
+from comments.models import CannedResponse
+
+
+def update(request, comment_pk, action):
```

Comment:
My code smell senses are tingling, but I am not able to offer a viable alternative. Thinking from a RESTful perspective, I would want each 'action' here to be handled by a separate endpoint and this would eliminate the 'if-elif' structure here.

If no change is made, should there be some tests to catch possible regressions in the future if someone has to edit this code?

Reply:
This code has been 'copied' from `wagtail_comments_xtd`. Will discuss in the call